File: cpp/Kode/Skill.cpp
#include "Skill.h"
#include "Type.h" 

Skill::Skill(std::string n, int mc, double cd, Type* t) : 
    Name(n), ManaCost(mc), Cooldown(cd), skillType(t) {}

void Skill::displaySkillDetails() const {
    std::cout << "  - [SKILL] Name: " << Name << ", Mana Cost: " << ManaCost << ", Cooldown: " << Cooldown << "s" << std::endl;
    if (skillType) {
        std::cout << "    Type Details: ";
        skillType->displayType(); 
        std::cout << std::endl;
    }
}

File: java/Kode/Skill.java
import java.util.UUID;

public class Skill {
    private String name;
    private int manaCost;
    private double cooldown;
    private Type skillType; // Association to Type
    private UUID id;

    public Skill(String name, int manaCost, double cooldown, Type skillType) {
        this.name = name;
        this.manaCost = manaCost;
        this.cooldown = cooldown;
        this.skillType = skillType;
        this.id = UUID.randomUUID(); 
    }

    public String getName() {
        return name;
    }

    public void displaySkillDetails() {
        System.out.println("  - [SKILL] Name: " + name + ", Mana Cost: " + manaCost + ", Cooldown: " + cooldown + "s");
        if (skillType != null) {
            System.out.print("    Type Details: ");
            skillType.displayType();
            System.out.println();
        }
    }
}